File: bin/functions/metrics.R
#' Format metric results
#'
#' @param dataset The name of the dataset the metric has been calculated for
#' @param method The name of the method the metric has been calculated for
#' @param integration The name of the integration the metric has been calculated
#' for
#' @param metric_type The type of the metric, either 'Integration',
#' 'Classification', 'Mapping' or 'Unseen'
#' @param metric The name of the metric that has been calculated
#' @param value The value of the calculated metric
#'
#' @return data.frame containing the formatted results
format_metric_results <- function(dataset, method, integration, metric_type, metric, value) {
    if (!(metric_type %in% c("Integration", "Classification", "Mapping", "Unseen"))) {
        stop("'metric_type' must be one of 'Integration', 'Classification', 'Mapping' or 'Unseen'")
    }

    if (value < 0 || value > 1) {
        stop("'score' must be between 0 and 1")
    }

    message("Formatting metric results...")
    data.frame(
        "Dataset"     = dataset,
        "Method"      = method,
        "Integration" = integration,
        "Type"        = metric_type,
        "Metric"      = metric,
        "Value"       = value
    )
}

File: bin/functions/metrics.py
def format_metric_results(dataset, method, integration, metric_type, metric, value):
    """
    Format metric results

    Parameters
    ----------
    dataset
        The name of the dataset the metric has been calculated for
    method
        The name of the method the metric has been calculated for
    integration
        The name of the integration the metric has been calculated for
    metric_type
        The type of the metric, either 'IntegrationBatch', 'IntegrationBio',
        'Classification', 'Mapping' or 'Unseen'
    metric
        The name of the metric that has been calculated
    value
        The value of the calculated metric

    Returns
    -------
    DataFrame containing the formatted results
    """

    from pandas import DataFrame

    if not metric_type in [
        "IntegrationBatch",
        "IntegrationBio",
        "Classification",
        "Mapping",
        "Unseen",
    ]:
        raise ValueError(
            "'metric_type' must be one of 'IntegrationBatch', 'IntegrationBio', 'Classification', 'Mapping' or 'Unseen'"
        )

    if metric ==  "NA":
        from warnings import warn
        warn(f"Storing missing score for metric '{metric}'")
    else:
        if value < 0 or value > 1:
            raise ValueError("'score' must be between 0 and 1")

    print("Formatting metric results...")
    results = DataFrame(
        [
            {
                "Dataset": dataset,
                "Method": method,
                "Integration": integration,
                "Type": metric_type,
                "Metric": metric,
                "Value": value,
            }
        ]
    )

    return results

